### ISSUE TITLE
[Package Request] - strongswan

### ISSUE THREAD

**Issue by bplessis-swi** [amazonlinux/amazon-linux-2023#301]
**What package is missing from Amazon Linux 2022? Please describe and include package name.**

The strongswan package is missing

**Is this an update to existing package or new package request?**

New package

**Is this package available in Amazon Linux 2? If it is available via external sources such as EPEL, please specify.**

Available in AmazonLinux 2 via EPEL

**Any additional information you'd like to include. (use-cases, etc)**

Useful to maintain IPsec VPNs access with special constraint (ip-space over-lap, partial access through firewall or local proxy service)

**Comment by Criggie** [comment]
libreswan was in AL2, but they've chosen to take it out. 

VPNs can be terminated by the AWS service "VPNConnections"  so having it easily available as a package may be reducing the usage of that service.

I'd like the same - any of the *swan packages compiled for AWS Linux 2023, so I don't have to faff with self-packaging.

**Comment by florianlocqueneux** [comment]
AWS service "VPNConnections"  is not covering a lot of network functionalities like NAT / Double Nating.

**Comment by daniejstriata** [comment]
You can try these:
[libreswan and ldns x86_64.zip](https://github.com/amazonlinux/amazon-linux-2023/files/11544511/libreswan.and.ldns.x86_64.zip)

**TL;DR**
You can build the following source packages on a AL2023 host:
` https://kojipkgs.fedoraproject.org//packages/ldns/1.8.1/7.fc37/src/ldns-1.8.1-7.fc37.src.rpm`
` https://kojipkgs.fedoraproject.org//packages/libreswan/4.11/1.fc37/src/libreswan-4.11-1.fc37.src.rpm`
to build `libreswan `you need `ldns-devel` which is not found in AL2023.

**Comment by stewartsmith** [comment]
> Libreswan has been removed from AL2023 packages list (https://docs.aws.amazon.com/linux/al2023/release-notes/removed-packages.html). However, as AL2023 is based on Fodera, I though I 'll be able to install it with the package libreswan-4.11-1.fc38.x86_64.rpm where the package is available for download from Fodera Project website (https://download-ib01.fedoraproject.org/pub/fedora/linux/updates/38/Everything/x86_64/Packages/l/libreswan-4.11-1.fc38.x86_64.rpm ).

As per https://docs.aws.amazon.com/linux/al2023/ug/relationship-to-fedora.html 
> The Generally Available (GA) version of AL2023 isn't directly comparable to any specific Fedora release. The AL2023 GA version includes components from Fedora 34, 35, and 36. Some of the components are the same as the components in Fedora and some are modified. Other components more closely resemble the components in CentOS 9 Streams or were developed independently. The Amazon Linux kernel is sourced from the long-term support options that are on kernel.org, chosen independently from Fedora.

So it's not terribly unexpected that rebuilding packages from Fedora can work, trying to install them directly is quite likely not to.

**Comment by Slapy06** [comment]
Thank you @stewartsmith / @daniejstriata for your comment. I will update this topic soon...

**Comment by emerchalegre-mk** [comment]
Is there no way to install EPEL on Amazon Linux 2023?

**Comment by bplessis-swi** [comment]
> Is there no way to install EPEL on Amazon Linux 2023?

i don't think the issue is at that level, since they hacked a release which is not a direct descendant of any redhat release, EPEL would need to have dedicated builds for this distribution.

It doesn't look like they have even discussed of the subject on the devel mailing list

**Comment by waltercool** [comment]
Just wondering if the main reason is to boost their own VPN service?

**Comment by ozbenh** [comment]
I can assure you this is not a factor in our decisions. This request is still under evaluation

**Comment by stewartsmith** [comment]
> Just wondering if the main reason is to boost their own VPN service?

https://github.com/amazonlinux/amazon-linux-2023/issues/123 is the request to package the client for the AWS service, as it hasn't been packaged for Amazon Linux.

**Comment by benpbolton** [comment]
Can we confirm the request is still under evaluation?

**Comment by stewartsmith** [comment]
I can confirm it's still under consideration, yes.

**Comment by dvmrry** [comment]
You guys should talk to your AWS teams more -we were recommended to use EC2-based VPN from Local Zones to a CGW > TGW to avoid our LZs routing halfway back across the country.

Hell even the documentation suggests this - https://docs.aws.amazon.com/local-zones/latest/ug/local-zones-connectivity-ec2-vpn.html

**Comment by d0u9** [comment]
Just upgrade from AL2 to AL2023, and surprised that `libreswan` has gone.

**Comment by margussipria** [comment]
@stewartsmith is strongswan still under consideration after libreswan was added with https://docs.aws.amazon.com/linux/al2023/release-notes/relnotes-2023.3.20231218.html .

would like to still use strongswan configurations but also get patches from Amazon.

**Comment by terhosiikanen-nexi** [comment]
Strongswan is vital to our business, the AWS managed service(s) do not support our use cases, including the features already mentioned here.

**Comment by ozbenh** [comment]
Is `libreswan` not sufficient for your use case ?

**Comment by terhosiikanen-nexi** [comment]
> Is `libreswan` not sufficient for your use case ?

We are currently evaluating if the other side of the connection is able to change certain parameters, which would make libreswan a viable option.

**Comment by terhosiikanen-nexi** [comment]
Unfortunately we were unable to get Libreswan working with some of our partners using ikev2 + **left** subnet space defined by the **right** side + SNAT. We tried pretty much every combination of Libreswan parameters as well as PRE- and POST routing setups to no avail.

We simply didn't receive any responses back to our UDP encapsulated ESP packets, even though after debug-decrypting the packages looked exactly the same as with Strongswan (using tcpdump and wireshark). Strongswan worked fine straight out of the box.

Libreswan worked fine with some of the more simple setups though, but we don't want to run two different clients.

**Comment by Tristan-Lynass** [comment]
@stewartsmith @ozbenh With AL2 going into EOL in 2026, is StrongSwan still under consideration? Can you share why it was dropped?

**Comment by pikkemans** [comment]
I want to put it out there for others. We've worked around this by running strongswan in a docker container (Alpine) on our Amazon Linux 2023 instances. It's actually running pretty well. 

Of course some downsides as having to manage the docker image etc.

**Comment by tchesley85** [comment]
Hi All, are there any plans to return to this topic? I'm also trying to use strongSwan on AL23.

**Comment by ozbenh** [comment]
@Tristan-Lynass It wasn't dropped, afaik it was never in AL2 ... or am I not looking in the right place ? The team owning these packages at the time decided that `libreswan` was a better option. I will forward the concerns expressed here and see if they can come up with ways to make the functionalities mentioned here work (or decide to carry strongswan).

**Comment by stenlarsson** [comment]
> It wasn't dropped, afaik it was never in AL2 ... or am I not looking in the right place ?

As mentioned in the description, it was available via EPEL.

**Comment by ozbenh** [comment]
EPEL is not and never was a proper substitute for a package. EPEL stuff is completely unsupported with no guarantee of CVE patching, and even on AL2, may or may not work. It is fundamentally no better than someone doing a COPR build and sharing it with the world.

**Comment by stenlarsson** [comment]
Yes, except "someone" in this case is the Fedora project maintainers. Would be great if it was a proper Amazon Linux package, but EPEL is the next best thing.

**Comment by LAmb-DEX** [comment]
Our system uses L2TP/ipsec (ikev1) with libreswan in AL2. I am trying to set up the same in AL2023 because of AL2 end of support. But while I was able stablish ipsec connection I cannot move to the next step. 

Apparently AL2023 does not have the ppp modules. Have someone managed to use AL2023 for ipsec/l2tp VPN connection? 
It has been a learning by doing process so I hope I expressed it correctly

**Comment by cjmorri1** [comment]
I have built strongSwan 6.0.2 for AL2023 for x86_64 into an RPM.... https://download.copr.fedorainfracloud.org/results/cmorrison/strongSwan-AL2023/amazonlinux-2023-x86_64/09500240-strongswan/

**Comment by ConnySeidelAWS** [comment]
Package was added to SPAL and will be in one of the next releases.

**Comment by ConnySeidelAWS** [comment]
Package is now available in SPAL:
```
$ yum info strongswan.aarch64
Last metadata expiration check: 0:55:33 ago on Fri Mar 20 09:52:58 2026.
Available Packages
Name         : strongswan
Version      : 5.9.11
Release      : 1.2.spal2023
Architecture : aarch64
Size         : 1.6 M
Source       : strongswan-5.9.11-1.2.spal2023.src.rpm
Repository   : amazonlinux-spal
Summary      : An OpenSource IPsec-based VPN and TNC solution
URL          : https://www.strongswan.org/
License      : GPLv2+
Description  : The strongSwan IPsec implementation supports both the IKEv1 and IKEv2 key
             : exchange protocols in conjunction with the native NETKEY IPsec stack of the
             : Linux kernel.
```
